include/roboteam_ai/treeinterp/OffensiveStrategy.h

```diff
@@ -0,0 +1,57 @@
+//
+// Created by jesse on 14.10.19.
+//
+#ifndef RTT_TREE_PROTO_TYPE_H
+#define RTT_TREE_PROTO_TYPE_H
+
+#include <include/roboteam_ai/bt/tactics/DefaultTactic.h>
+#include "bt/BehaviorTree.hpp"
+#include "bt/Role.h"
+
+
+
+namespace bt {
+
+    class OffensiveStrategy {
+    public:
+        /**
+         * Constructor that initializes the robots vector
+         */
+        OffensiveStrategy();
+        /**
+         * Creates a Strategy behaviour tree. This tree is created in BTFactory. The tree that is created is also given to the OffensiveStrategy object, so it can be traced and adjusted.
+         * @return the behaviour tree that contains this strategy
+         */
+        std::shared_ptr<bt::BehaviorTree> createOffensiveStrategy();
+
+    private:
+        /**
+         * Vector that is used by RobotDealer to determine how to match robots to robot IDs.
+         */
+        std::vector<std::pair<std::string, rtt::ai::robotDealer::RobotType>> robots;
+
+        /**
+         * The tree that is given to the offensive strategy
+         */
+        std::shared_ptr<bt::BehaviorTree> tree;
+
+        /**
```

Comment:
So what kind of design pattern are you really choosing now. Are you going to make everything like this with ownership of the tree or not? This is inconsistent with what you did in both of the new "Role" files.

Reply:
I'mtalking about the std::shared_ptr<bt::BehaviorTree tree> bit.